class.c

```diff
@@ -1504,6 +1504,35 @@ class_descendants(VALUE klass, bool immediate_only)
  *     A.subclasses        #=> [D, B]
  *     B.subclasses        #=> [C]
  *     C.subclasses        #=> []
+ *
+ *
+ *  Note that unlike Module#ancestors, this method's result
+ *  is not a representation of some internal state but is
+ *  calculated dynamically. It might lead to a non-deterministic
```

Comment:
> is not a representation of some internal state but is calculated dynamically

It very much is a representation of some internal state. `Class` has basically a weaklist of references to it's subclasses.